app/assets/javascripts/miq_application.js

```diff
@@ -707,23 +707,10 @@ function miqAjaxAuth(url) {
     serialized: miqSerializeForm('login_div'),
   };
 
-  return vanillaJsAPI.login(credentials.login, credentials.password)
-    .then(function() {
-      return vanillaJsAPI.ws_init();
-    })
-    .then(function() {
-      return storeUserFeatures();
```

Comment:
So this one is a bit tricky..

We can't do that before `/dashboard/authenticate` with cookie auth,
so this needs to happen after,
but *before* the redirect.

And we don't have any such hooks in jquery_overrides, so we'd have to output something like..

```
storeUserFeatures().then(function() { ... the actual redirect code... });
```

from the ruby side.

Or possibly we can just make sure we clear the data (`delete localStorage.userFeatures;`),
and change any code that wants to use it (no such code seems to exist now) to call this on demand.

(Comes from https://github.com/ManageIQ/manageiq-ui-classic/pull/4756)

Reply:
Oh, the storeUserFeatures will be useful for the RBAC stuff (maybe) in the future. I don't really understand the way you explained where should I put it though :confused:

Reply:
Ah, yup, `miqOnLoad` could work :+1: 

(But that means you somehow have to make sure that `dashboard#show` outputs that `miqOnLoad` after being redirected to from `dashboard#authenticate`.)

(Or force it on every full screen load, and only do the request if not found `localStore`.)

I guess you'll want to do something like...

```js
if (localStorage.userFeatures) {
  ManageIQ.userFeaturesPromise = Promise.resolve(JSON.parse(localStorage.userFeatures));
  return;
}

ManageIQ.userFeaturesPromise = storeUserFeatures()
  .then(function() {
    return JSON.parse(localStorage.userFeatures);
  });
```

Reply:
Or just rewrite `storeUserFeatures` to something like..

```js
function storeUserFeatures() {
  var features = localStorage.userFeatures && JSON.parse(localStorage.userFeatures);

  if (features) {
    return Promise.resolve(features);
  }

  return http.get('/api?attributes=identity')
    .then(function(data) {
      var features = data.identity.miq_groups.find(function(group) {
        return data.identity.group === group.description;
      }).product_features;

      localStorage.userFeatures = JSON.stringify(features);
      return features;
    });
}
```

Reply:
(And don't forget to `delete window.localStorage.userFeatures;` when on the login screen.)

Reply:
Not being used yet, so we can indeed solve this by removing the call now.